tests/scrape.py

```diff
@@ -170,8 +170,17 @@ def fetch(url, data='', agent=None, referrer=None, charset=None, verbose=0,
     # Prepare the POST data.
     if not method:
         method = data and 'POST' or 'GET'
-    if data and not isinstance(data, str): # Unicode not allowed here
-        data = urlencode(data)
+
+    if not data:
+        data_str = ''
+    elif isinstance(data, str):
+        data_str = data
+    elif isinstance(data, unicode):
+        data_str = data.encode('utf-8')
+    elif isinstance(data, dict):
+        data_str = urlencode(data)
```

Comment:
Reading the code, urlquote(), used in urlencode(), supports both str and unicode. Added a comment.